apps/property-manager/app/modules/properties/property/assets/blocks/edit/index.tsx

```diff
@@ -0,0 +1,275 @@
+import { zodResolver } from '@hookform/resolvers/zod'
+import { ArrowLeft } from 'lucide-react'
+import { useEffect } from 'react'
+import { useForm } from 'react-hook-form'
+import { Link, useNavigate, useParams } from 'react-router'
+import { toast } from 'sonner'
+import { z } from 'zod'
+import { useGetPropertyBlock, useUpdatePropertyBlock } from '~/api/blocks'
+import { ErrorContainer } from '~/components/ErrorContainer'
+import { LoadingContainer } from '~/components/LoadingContainer'
+import { Button } from '~/components/ui/button'
+import { FieldGroup } from '~/components/ui/field'
+import {
+	Form,
+	FormControl,
+	FormDescription,
+	FormField,
+	FormItem,
+	FormLabel,
+	FormMessage,
+} from '~/components/ui/form'
+import { ImageUpload } from '~/components/ui/image-upload'
+import { Input } from '~/components/ui/input'
+import { Spinner } from '~/components/ui/spinner'
+import { Textarea } from '~/components/ui/textarea'
+import {
+	TypographyH2,
+	TypographyMuted,
+	TypographySmall,
+} from '~/components/ui/typography'
+import { useUploadObject } from '~/hooks/use-upload-object'
+import { safeString } from '~/lib/strings'
+import { cn } from '~/lib/utils'
+import { useProperty } from '~/providers/property-provider'
+
+const ValidationSchema = z.object({
+	name: z.string().min(1, 'Name is required'),
+	image_url: z
+		.string()
+		.url('Please upload an image')
+		.optional()
+		.or(z.literal('')),
+	description: z
+		.string()
+		.max(500, 'Description must be less than 500 characters')
+		.optional(),
+	status: z.enum(
+		[
+			'PropertyBlock.Status.Active',
+			'PropertyBlock.Status.Maintenance',
+			'PropertyBlock.Status.Inactive',
+		],
+		{ error: 'Please select a status' },
+	),
+})
+
+type FormSchema = z.infer<typeof ValidationSchema>
+
+const statusOptions: Array<{ label: string; value: FormSchema['status'] }> = [
+	{ label: 'Active', value: 'PropertyBlock.Status.Active' },
+	{ label: 'Inactive', value: 'PropertyBlock.Status.Inactive' },
+	{ label: 'Maintenance', value: 'PropertyBlock.Status.Maintenance' },
+]
+
+export function EditPropertyAssetBlocksModule() {
+	const { clientUserProperty } = useProperty()
+	const navigate = useNavigate()
+	const { blockId } = useParams()
+
+	const {
+		isPending: isLoadingData,
+		data,
+		error,
+	} = useGetPropertyBlock({
+		property_id: clientUserProperty?.property?.id ?? '',
+		id: blockId ?? '',
```

Comment:
there's a function in the system called `safeString`. Please use those instead when you're defaulting to empty string.

```suggestion
		property_id: safeString(clientUserProperty?.property?.id),
		id: safeString(blockId),
```